core/src/main/scala/org/apache/gearpump/util/FileServer.scala

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.util
+
+import java.io.File
+
+import akka.actor.{Actor, Stash}
+import akka.io.IO
+import org.apache.commons.httpclient.HttpClient
+import org.apache.commons.httpclient.methods.{ByteArrayRequestEntity, GetMethod, PostMethod}
+import org.apache.commons.io.FileUtils
+import spray.can.Http
+import spray.http.HttpMethods._
+import spray.http._
+
+import scala.util.{Failure, Success, Try}
+
+/**
+ *
+ * Should not use this to server too big files(more than 100MB), otherwise OOM may happen.
+ *
+ * port: set port to 0 if you want to bind to random port
+ */
+class FileServer(rootDir: File, host: String, port : Int) extends Actor with Stash {
```

Comment:
Currently it is still difficult for this FileServer to store/get file larger than 1GB. 

We still need to study spary to find way to support it.